### ISSUE TITLE
Unable to reference class FBSDKLoginButton

### ISSUE THREAD

**Issue by sebbean** [facebook/facebook-ios-sdk#750]
Works on simulator, doesn't work on device.

Errors:
` "_OBJC_CLASS_$_FBSDKLoginButton", referenced from:`

`ld: warning: ignoring file 
...
FBSDKLoginKit.framework/FBSDKLoginKit, 

file was built for archive which is not the architecture being linked (armv7): 

...
FBSDKCoreKit.framework/FBSDKCoreKit, 

*file was built for archive which is not the architecture being linked (armv7): *
...
FBSDKCoreKit.framework/FBSDKCoreKit
`

When building for device after adding fb-ios-sdk as submodule and adding FBSDKCoreKit and FBSDKLoginKit as sub-projects + linked frameworks:

![screen shot 2015-05-06 at 11 58 10 pm](https://cloud.githubusercontent.com/assets/423997/7508295/05b4c914-f44c-11e4-9d4f-3795a537aadc.png)
![screen shot 2015-05-06 at 11 57 53 pm](https://cloud.githubusercontent.com/assets/423997/7508296/05b8d702-f44c-11e4-8224-78235f22998d.png)

**Comment by chrisp-fb** [comment]
I created a new project and added the submodule via `git submodule add git@github.com:facebook/facebook-ios-sdk.git` then `git submodule update --init --recursive`. Then I dragged the CoreKit and LoginKit project files into my project and added them as dependencies for linking like your screenshot, and then added some code to reference FBSDK* classes. I was not able to reproduce your problem with Xcode 6.3 and was able to build for my iphone5s device as well as for various simulators.

Are you performing some other sort of build that would require the building of all architectures in debug (we do already disable the flag for release builds)?
